vivo-project/Vitro
# Vivo 2004

**[JIRA Issue VIVO-2004](https://jira.lyrasis.org/browse/VIVO-2004)**: (please link to issue)


# What does this pull request do?
Added a new http authentication API (http://.../vivo/robot-authentification) allowing a robot to perform a login and retrieve a return code 200 in case of success or 401 in case of an authentication error

# What's new?
        modified:   api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/edit/Authenticate.java
        created:     api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/edit/RobotAuthenticate.java
        modified :  api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/login/LoginProcessBean.java
        created:     api/src/test/java/edu/cornell/mannlib/vitro/webapp/controller/edit/RobotAuthenticateTest.java

LoginProcessBean.java:
Add a getter on Message to get the info contained by the loginBean

RobotAuthenticate.java
Extension of Authenticate.java to adjust the response to return the appropriate authentication code

Authenticate.java
Chagement of variables inherited by RobotAuthenticate from private to protected

RobotAuthenticateTest.java
Corresponding test case

# How should this be tested?

first method RobotAuthenticate.java to see 401 return code. can be compare with Authenticate.java TestCase

second method - using CURL

1- Start VIVO (suppose on http://localhost:8080/vivo)
2- Run this CURL
```
curl --location --request POST 'http://localhost:8080/vivo/robot_authenticate' \
--header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8' \
--header 'Accept-Language: fr,fr-FR;q=0.8,en-US;q=0.5,en;q=0.3' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--header 'Origin: http://localhost:8080' \
--header 'Connection: keep-alive' \
--header 'Referer: http://localhost:8080/vivo/' \
--header 'Upgrade-Insecure-Requests: 1' \
--data-urlencode 'loginName=YOUR_NAME' \
--data-urlencode 'loginPassword=YOUR_PASSWORD' \
--data-urlencode 'loginForm=Log+in'
```

* Expected result

running on  'http://localhost:8080/vivo/robot_authenticate' or  'http://localhost:8080/vivo/authenticate' gives the same result for a valid login

For not valid login it give someting like:
`<!doctype html><html lang="fr"><head><title>État HTTP 401 – Non authorisé</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>État HTTP 401 – Non authorisé</h1><hr class="line" /><p><b>Type</b> Rapport d'état</p><p><b>message</b> The email or password you entered is incorrect.</p><p><b>description</b> La requête nécessite une authentification HTTP.</p><hr class="line" /><h3>Apache Tomcat/9.0.45</h3></body></html>heon@LinuxUQAM[70]:VS-GIT$ `

look the "HTTP 401 "

# Additional Notes:
N/A
Example:
N\A
# Interested parties
@brianjlowe
